### PR TITLE
fix(firebase_storage): updated `firebase_storage_web` to allow  `http` package dependency range '>=0.13.0 <2.0.0'. Fixes dependency resolution.

### LINKED ISSUE #11073: 🐛 [firebase_storage] Version solving failed for most recent http version (v1.0.0)
## Bug report

**Describe the bug**
Firebase storage requires an older version of the [http package](https://pub.dev/packages/http) that was recently released. Causing a version-solving failed error when trying to upgrade.

### Steps to reproduce

1. Create a new Flutter app
2. Add the most recent versions of Firebase Storage (v11.2.2) and http (v1.0.0) packages
3. Run `flutter pub get`
4. See error for dependency version conflict

### Expected behavior

Should be able to use the latest versions of both packages

### Sample project

N/A

---

## Additional context

N/A

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.22621.1778], locale en-US)
    • Flutter version 3.10.0 on channel stable at C:\dev\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (3 weeks ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\Preston\AppData\Local\Android\sdk
    • Platform android-33-ext5, build-tools 33.0.2
    • Java binary at: C:\Users\Preston\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\222.4459.24.2221.9971841\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.26)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.11.33529.622
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 2022.2)
    • Android Studio at C:\Users\Preston\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\222.4459.24.2221.9971841
    • Flutter plugin version 73.0.1
    • Dart plugin version 221.6103.1
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)

[✓] VS Code (version 1.78.2)
    • VS Code at C:\Users\Preston\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.64.0

[✓] Connected device (3 available)
    • Pixel 4 XL (mobile) • 9B071FFBA003B1 • android-arm64  • Android 13 (API 33)
    • Windows (desktop)   • windows        • windows-x64    • Microsoft Windows [Version 10.0.22621.1778]
    • Edge (web)          • edge           • web-javascript • Microsoft Edge 113.0.1774.57

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
- adaptive_dialog 1.9.0-no-macos.2 [animations collection dynamic_color flutter intersperse meta]
- cached_network_image 3.2.3 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- cloud_firestore 4.8.0 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cloud_firestore_odm 1.0.0-dev.60 [cloud_firestore flutter json_annotation meta]
- collection 1.17.1
- connectivity_plus 4.0.1 [flutter flutter_web_plugins connectivity_plus_platform_interface js meta nm]
- cupertino_icons 1.0.5
- easy_localization 3.0.2 [flutter shared_preferences intl args path easy_logger flutter_localizations]
- firebase_analytics 10.4.2 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_app_check 0.1.4+2 [firebase_app_check_platform_interface firebase_app_check_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 4.6.2 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.13.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 3.3.2 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_dynamic_links 5.3.2 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- firebase_messaging 14.6.2 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_storage 11.2.2 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- flutter 0.0.0 [characters collection js material_color_utilities meta vector_math sky_engine]
- flutter_easyloading 3.0.5 [flutter flutter_spinkit]
- flutter_form_builder 9.0.0 [flutter intl collection]
- flutter_local_notifications 14.1.1 [clock flutter flutter_local_notifications_linux flutter_local_notifications_platform_interface timezone]
- flutter_native_splash 2.3.0 [args flutter flutter_web_plugins js html image meta path universal_io xml yaml]
- flutter_picker 2.1.0 [flutter]
- flutter_riverpod 2.3.6 [collection flutter meta riverpod state_notifier]
- flutter_secure_storage 8.0.0 [flutter flutter_secure_storage_linux flutter_secure_storage_macos flutter_secure_storage_platform_interface flutter_secure_storage_web flutter_secure_storage_windows meta]
- flutter_svg 2.0.6 [flutter vector_graphics vector_graphics_codec vector_graphics_compiler]
- form_builder_cupertino_fields 0.2.0 [flutter flutter_form_builder intl]
- form_builder_phone_field 1.4.0 [flutter flutter_form_builder phone_number country_pickers]
- form_builder_validators 9.0.0 [flutter flutter_localizations intl]
- freezed_annotation 2.2.0 [collection json_annotation meta]
- geocoding 2.1.0 [flutter geocoding_platform_interface geocoding_android geocoding_ios]
- geolocator 9.0.2 [flutter geolocator_platform_interface geolocator_android geolocator_apple geolocator_web geolocator_windows]
- go_router 7.1.1 [collection flutter flutter_web_plugins logging meta]
- goda_device_protocol 0.31.0 [flutter convert fl_chart synchronized logger retry mutex flutter_blue_plus flutter_lints]
- http 0.13.6 [async http_parser meta]
- image_cropper 4.0.1 [flutter image_cropper_platform_interface image_cropper_for_web]
- image_gallery_saver 2.0.1 [flutter]
- image_picker 0.8.7+5 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_platform_interface]
- internet_connection_checker 1.0.0+1
- json_annotation 4.8.1 [meta]
- local_auth 2.1.6 [flutter local_auth_android local_auth_ios local_auth_platform_interface local_auth_windows]
- logger 1.3.0
- overlay_support 2.1.0 [flutter async]
- package_info_plus 4.0.2 [ffi flutter flutter_web_plugins http meta path package_info_plus_platform_interface win32]
- path_provider 2.0.15 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- permission_handler 10.2.0 [flutter meta permission_handler_android permission_handler_apple permission_handler_windows permission_handler_platform_interface]
- riverpod_annotation 2.1.1 [meta riverpod]
- screenshot 2.1.0 [flutter]
- sentry_flutter 7.6.3 [flutter flutter_web_plugins sentry package_info_plus meta]
- sentry_logging 7.6.3 [logging sentry]
- shared_preferences 2.1.1 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows```

</details>

---

### PR DESCRIPTION
**Pull request by OutdatedGuy**
## Description

Updated the [http](https://pub.dev/packages/http) package to `v1.0.0`. This will make the plugin compatible with all the packages that use the latest `v1.0.0` of the http package.

## Related Issues

Fixes #11073

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by OutdatedGuy** [comment]
> Seems like there is a clear domino effect taking place, tests are failing because they cant install the packages to test because of the version conflict
> 
> Quite ironic

Yep, as all packages use pub reference instead of path reference. Updating any major dependency seems to fails all the tests.

Unfortunately I don't know how to overcome that issue.

File: packages/firebase_storage/firebase_storage_web/pubspec.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ dependencies:
     sdk: flutter
   flutter_web_plugins:
     sdk: flutter
-  http: ^0.13.0
+  http: ^1.0.0
```
Comment (felangel) [inline packages/firebase_storage/firebase_storage_web/pubspec.yaml:21]:
```suggestion
  http: '>=0.13.0 <2.0.0'
```

Comment (OutdatedGuy) [reply packages/firebase_storage/firebase_storage_web/pubspec.yaml]: Thanks for the suggestion @felangel

**Comment by sparrow001** [comment]
Does this mean this will be pushed soon???? :0

**Review by felangel — APPROVED** [review]
LGTM seems like the one failing test could be flaky so I’d try re-running it.

**Review by russellwheatley — APPROVED** [review]

**Review by lesnitsky — APPROVED** [review]

**Comment by OutdatedGuy** [comment]
Why is [this test](https://github.com/firebase/flutterfire/actions/runs/5370589685/jobs/9751458465) still running even after 5 hours?

Screenshot 2023-06-26 at 6 47 21 PM

**Comment by sparrow001** [comment]
Seems like a good old GitHub bug. Its stuck on terminating all the processes, so its safe to assume the actual test was successful at least.

**Comment by russellwheatley** [comment]
> Why is [this test](https://github.com/firebase/flutterfire/actions/runs/5370589685/jobs/9751458465) still running even after 5 hours? @russellwheatley @lesnitsky
> 
> Screenshot 2023-06-26 at 6 47 21 PM

No idea, it passed but never left the "in progress" state. Just a bug in Github Actions somewhere.

**Pull request merged**